core/src/avm2/globals/flash/utils/bytearray.rs

```diff
@@ -0,0 +1,939 @@
+        bytearray.write_bytes(string.as_bytes());
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn compress<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut bytearray) = this
+        .unwrap()
+        .as_bytearray_mut(activation.context.gc_context)
+    {
+        if let Value::String(string) = args.get(0).unwrap_or(&Value::Undefined) {
+            let compressed = match string.as_str() {
+                "zlib" => bytearray.zlib_compress(),
+                "deflate" => bytearray.deflate_compress(),
+                &_ => return Ok(Value::Undefined),
+            };
+            if let Ok(buffer) = compressed {
+                bytearray.clear();
+                bytearray.write_bytes(&buffer);
+            }
+        }
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn uncompress<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut bytearray) = this
+        .unwrap()
+        .as_bytearray_mut(activation.context.gc_context)
+    {
+        if let Value::String(string) = args.get(0).unwrap_or(&Value::Undefined) {
+            let compressed = match string.as_str() {
+                "zlib" => bytearray.zlib_decompress(),
+                "deflate" => bytearray.deflate_decompress(),
+                &_ => return Ok(Value::Undefined),
+            };
+            if let Ok(buffer) = compressed {
+                bytearray.clear();
+                bytearray.write_bytes(&buffer);
+            }
+        }
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn deflate<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut bytearray) = this
+        .unwrap()
+        .as_bytearray_mut(activation.context.gc_context)
+    {
+        if let Ok(buffer) = bytearray.deflate_compress() {
+            bytearray.clear();
+            bytearray.write_bytes(&buffer);
+        }
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn inflate<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut bytearray) = this
+        .unwrap()
+        .as_bytearray_mut(activation.context.gc_context)
+    {
+        if let Ok(buffer) = bytearray.deflate_decompress() {
+            bytearray.clear();
+            bytearray.write_bytes(&buffer);
+        }
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn create_class<'gc>(mc: MutationContext<'gc, '_>) -> GcCell<'gc, Class<'gc>> {
+    let class = Class::new(
+        QName::new(Namespace::package("flash.utils"), "ByteArray"),
+        Some(QName::new(Namespace::public(), "Object").into()),
+        Method::from_builtin(instance_init),
+        Method::from_builtin(class_init),
+        mc,
+    );
```

Comment:
ByteArray class should be non-dynamic.
`class.write(mc).set_attributes(ClassAttributes::SEALED);`